tests/main/interfaces-timeserver-control/task.yaml

```
@@ -56,6 +56,16 @@ execute: |
         retry --wait 5 sh -c 'test "$(busctl get-property org.freedesktop.timedate1 /org/freedesktop/timedate1 org.freedesktop.timedate1 NTP)" = "b '"$value"'"'
     done
 
+    # Only test timesync related functonality when available (systemd >= 239).
+    if busctl tree org.freedesktop.timesync1 | MATCH "/org/freedesktop/timesync1"; then
+        echo "Check dbus access to org.freedesktop.timesync1 properties"
+        serverName=$(busctl get-property --system org.freedesktop.timesync1 /org/freedesktop/timesync1 org.freedesktop.timesync1.Manager ServerName | awk -F'"' '{print $2}')
+
+        echo "Check that timedatectl timesync commands can be used"
```

Might as well add a simple check that `timedatectl` works without subcommand.

```suggestion
        echo "Check that timedatectl timesync commands can be used"
        test-snapd-timedate-control-consumer.timedatectl-timeserver
```

---

At this point timedatectl-timeserver was already used, and we are specifically interested in testing the timesync commands. Do you feel its plausible that it works with sub-command and not without?

---

Without a subcommand it prints a status, which could theoretically (though highly unlikely) require system access that the subcommand doesn't use. Just for completeness, could ensure that the subcommand-free command works, since the customer explicitly said they call it. It's probably fine either way, though.

---

Thanks, saw the reply now, added.